dask_expr/_collection.py

```diff
@@ -3546,10 +3548,37 @@ def to_numeric(arg, errors="raise", downcast=None, meta=None):
     )
 
 
-def to_datetime(arg, **kwargs):
-    if not isinstance(arg, FrameBase):
-        raise TypeError("arg must be a Series or a DataFrame")
-    return new_collection(ToDatetime(frame=arg, kwargs=kwargs))
+def to_datetime(arg, meta=None, **kwargs):
+    tz_kwarg = {"tz": "utc"} if kwargs.get("utc") else {}
+
+    (arg,) = _maybe_from_pandas([arg])
+
+    if meta is None:
+        if isinstance(arg, Index):
+            meta = get_meta_library(arg).DatetimeIndex([], **tz_kwarg)
+            meta.name = arg.name
+        elif not (is_dataframe_like(arg) or is_series_like(arg)):
+            raise NotImplementedError(
+                "dask.dataframe.to_datetime does not support "
+                "non-index-able arguments (like scalars)"
+            )
+        else:
+            meta = meta_series_constructor(arg)([pd.Timestamp("2000", **tz_kwarg)])
+            meta.index = meta.index.astype(arg.index.dtype)
+            meta.index.name = arg.index.name
+    if PANDAS_GE_200 and "infer_datetime_format" in kwargs:
```

Comment:
Lets deprecate in dask/dask, I don't want this stuff in here if we can avoid it

Reply:
I'd also be ok with just ignoring it

Reply:
This is already deprecated in `dask/dask`. I'll remove the handling.